src/services/auth.service.ts

```diff
@@ -6,12 +6,14 @@ import { ITokenPayload } from 'src/interfaces/auth/auth.interface';
 
 @Service()
 export class AuthorizationService {
+  constructor(
+    private readonly jwt: JWTService,
+    private readonly redis: RedisService
+  ) { }
   private static instance: AuthorizationService;
-  private static jwt = Container.get(JWTService);
-  private static redis = Container.get(RedisService);
   public static getInstance(): AuthorizationService {
```

Comment:
Why are you declaring the getInstance method, why are we declaring the auth service as a singleton?

Reply:
All Services are treated like [Singleton with the @Service() decorator by default](https://github.com/typestack/typedi/blob/develop/docs/typescript/02-basic-usage-guide.md#singleton-vs-transient-classes). The getInstance method is only for developers to realize at first sight the pattern is being used. Maybe it is not neccesary to use the this.instance and the getInstance() method, and instead doing a Container.get(AuthorizationService) in the files that it is imported.
Regarding to the @Service() decorator, it is because we need it to be injected in the Container with its dependencies ( JWTService and RedisService in this case )